spec/test_runner_spec.rb

```diff
@@ -4,29 +4,77 @@
   let(:runner) { JavaTestHook.new }
 
   describe '#run' do
-    context 'when test does not compile' do
-      let(:file) { File.new 'spec/data/errored/SubmissionTest.java' }
-      let(:result) { runner.run!(file) }
+    let(:file) { runner.compile(treq(content, test, extra)) }
+    let(:raw_results) { runner.run!(file) }
+    let(:results) { raw_results[0] }
 
-      it { expect(result[1]).to eq :errored }
-      it { expect(result[0]).to include 'error: reached end of file while parsing' }
+    let(:extra) { '' }
+    let(:content) { '' }
+    let(:test) do
+        <<JAVA
+  @Test
+  public void testGetAnInt() {
+    Assert.assertEquals(3, new Foo().getAnInt());
+  }
+JAVA
+      end
 
+    context 'when test does not compile' do
+      let(:content) do
+        <<JAVA
+class Foo {
+  public int getAnInt() {
+    return 3;
+  }
+JAVA
+      end
+      it { expect(results).to include 'error: reached end of file while parsing' }
     end
 
     context 'when test fails' do
-      let(:file) { File.new 'spec/data/failed/SubmissionTest.java' }
-      let(:result) { runner.run!(file) }
-
-      it { expect(result[1]).to eq :failed }
-      it { expect(result[0]).to include 'There was 1 failure' }
+      let(:content) do
+        <<JAVA
+class Foo {
+  public int getAnInt() {
+    return 2;
+  }
+}
+JAVA
+      end
+      it { expect(results).to eq [['testGetAnInt', :failed, 'expected:<3> but was:<2>']] }
 
     end
 
     context 'when test passes' do
-      let(:file) { File.new 'spec/data/passed/SubmissionTest.java' }
-      let(:result) { runner.run!(file) }
+          let(:test) do
+        <<JAVA
+  @Test
+  public void testGetAnInt() {
+    Assert.assertEquals(3, new Foo().getAnInt());
+  }
+  @Test
+  public void testGetAFoo() {
+    Assert.assertEquals(3, 4);
+  }
+  @Test
+  public void testGetACharacter() {
+    Assert.assertEquals('a', 'a');
+  }
+
+JAVA
+      end
+
+      let(:content) do
+        <<JAVA
+class Foo {
+  public int getAnInt() {
+    return 3;
+  }
+}
+JAVA
+      end
 
-      it { expect(result[1]).to eq :passed }
+      it { expect(results).to eq [['testGetAnInt', :passed, nil], ["testGetACharacter", :passed, nil], ['testGetAFoo', :failed, 'expected:<3> but was:<4>']] }
```

Comment:
I'd think we need more test cases here, that test **failing** scenarios where the following types are used: 

  * Strings
  * Chars
  * Arrays or collections of strings and chars.